lib/builtin-models.js

```diff
@@ -52,8 +55,33 @@ module.exports = function(registry) {
     require('../common/models/checkpoint.js'));
 
   function createModel(definitionJson, customizeFn) {
+    // Clone the JSON definition to allow applications
+    // to modify model settings while not affecting
+    // settings of new models created in the local registry
+    // of another app.
+    // This is needed because require() always returns the same
+    // object instance it loaded during the first call.
+    definitionJson = cloneDeepJson(definitionJson);
+
     var Model = registry.createModel(definitionJson);
     customizeFn(Model);
     return Model;
   }
 };
+
+// Because we are cloning objects created by JSON.parse,
+// the cloning algorithm can stay much simpler than a general-purpose
+// "cloneDeep" e.g. from lodash.
+function cloneDeepJson(obj) {
```

Comment:
We are trying to avoid using `lodash` in our runtime code in order to keep the size of browserified bundle small (see https://github.com/strongloop/loopback/issues/989). lodash contributes about 350kb unminified and increases the bundle size by approx 20% - see https://github.com/strongloop/loopback-datasource-juggler/pull/538.